src/main/java/vct/col/util/AbstractTypeCheck.java

```diff
@@ -1158,6 +1158,15 @@ public void visit(OperatorExpression e) {
         e.setType((Type) ((PrimitiveType) t).firstarg());
         break;
       }
+      case OptionGetOrElse: {
+        if (!tt[0].isPrimitive(PrimitiveSort.Option)) {
+          Fail("first argument is %s rather then an option ", tt[0]);
+        } else if (!tt[1].comparableWith(source(), (Type) tt[0].firstarg())) {
+          Fail("type of the second argument %s does not match the value type of the option %s", tt[1], tt[0].firstarg());
+        }
+        e.setType(tt[1]);
```

Comment:
why not tt[0].firstarg()? let's say that we actually get a value from the Option parameter, then I would rather keep the original type than the one I use as fallback. Of course these may be the same (I am actually not sure if currently there is another possibility) but in the case there is or that it will be, then I would rather stay with the original type. Or am I misunderstanding what is happening here?

Reply:
Hmm I see your point. For example if we have an option `o1` with a float (currently not supported, but suppose), and I say `getOrElseOption(o1, 2)` where 2 is an integer, the type would become int instead of float. I will change this to `tt[0].firstarg()` (in the next commit).

Reply:
Just my 2ct: the correct solution is to pick the least common supertype of `tt[0].firstarg()` and `tt[1]`, e.g. `getOrElseOption(Some(object), null)` should prefer `tt[0].firstarg()` whereas `getOrElseOption(Some(superclassInstance), subclassInstance)` should prefer `tt[1]`. VerCors can't do that though, so `tt[0].firstarg()` is the better solution I think.